### PR TITLE
Client updates

### PR DESCRIPTION
**Pull request by dshybeka**
1. Externalized io pool size, max connections per route and total max connections. They can be set via reportportal.properties now:
 ```
rp.io.pool.size=100
rp.max.connections.per.route=50
rp.max.connections.total=100
```
2. Update handling of cases when RP returns html instead of regular json: added new serializer, which run last, if media type is not json the serializer will throw an exception with error message.

3. Added handling of success/error/complete for all .subscribe methods. This will help us to avoid OnErrorNotImplementedException. Restendpoint repo has been updated in similar way as well. So this particular merge require Restendpoint updates(https://github.com/reportportal/restendpoint/commit/0f0be2c482f9f3d8695c05ce7bc67c5bfba7fd5a  forgot to create separated branch :( ) to be released first.

plus probably it make sense to update RP UI(user profile page) to mach latest available properties.

### CODE REVIEW

File: src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.epam.reportportal.serializers;
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java]: apache 2.0 licence header

File: src/main/java/com/epam/reportportal/listeners/ListenerParameters.java [hunk 2]
```diff
@@ -68,6 +82,10 @@ public ListenerParameters(PropertiesLoader properties) {
 		this.keystore = properties.getProperty(KEYSTORE_RESOURCE);
 		this.keystorePassword = properties.getProperty(KEYSTORE_PASSWORD);
 		this.rerun = properties.getPropertyAsBoolean(RERUN, false);
+
+		this.ioPoolSize = properties.getPropertyAsInt(IO_POOL_SIZE, 100);
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/listeners/ListenerParameters.java]: let's make this values as constants

File: src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	@Override
+	public <T> byte[] serialize(T t) throws SerializerException {
+
+		throw new NotImplementedException();
+	}
+
+	@Override
+	public <T> T deserialize(byte[] content, Class<T> clazz) throws SerializerException {
+
+		throw new InternalReportPortalClientException("Report portal is not functioning correctly. Response is not json");
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java]: i guess the better place for this logic is ErrorHandler, not a serializer. Can we try to move it there?

File: src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
+package com.epam.reportportal.utils;
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java]: apache 2.0 licence header

File: src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+import io.reactivex.CompletableObserver;
+import io.reactivex.FlowableSubscriber;
+import io.reactivex.MaybeObserver;
+import io.reactivex.disposables.Disposable;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SubscriptionUtils {
```
Comment (avarabyeu) [inline src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java]: `@author`

File: src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java (hunk 3 above)
Comment (dshybeka) [reply src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java]: I was going to do the same, but the thing is that as I understood(tried) ErrorHandler is used when server responded with error. In this case server responded with 200 status but data was wrong, so we need Serializer to handle it. What do you think ?

File: src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java (hunk 5 above)
Comment (dshybeka) [reply src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java]: added

File: src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java (hunk 4 above)
Comment (dshybeka) [reply src/main/java/com/epam/reportportal/utils/SubscriptionUtils.java]: added

File: src/main/java/com/epam/reportportal/listeners/ListenerParameters.java (hunk 2 above)
Comment (dshybeka) [reply src/main/java/com/epam/reportportal/listeners/ListenerParameters.java]: sure

File: src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java (hunk 1 above)
Comment (dshybeka) [reply src/main/java/com/epam/reportportal/serializers/NotJsonSerializer.java]: added

**Pull request merged**